### PR TITLE
#1 :tada: First commit to public, open source repository

### PR DESCRIPTION
**Pull request by d-ryan-ashcraft**

### CODE REVIEW

File: orphedomos-test/orphedomos-test-no-buildkit/pom.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  <name>Orphedomos::Test::No BuildKit</name>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.technologybrewery.orphedomos</groupId>
+        <artifactId>orphedomos-maven-plugin</artifactId>
+        <configuration>
+          <imageName>pjablonski/orphetest</imageName>
```
Comment (peter-mcclonski) [inline orphedomos-test/orphedomos-test-no-buildkit/pom.xml]: Can we anonymize this?

File: orphedomos-test/orphedomos-test-multimodule/pom.xml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        <module>orphedomos-test-multimodule-b</module>
+    </modules>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.technologybrewery.orphedomos</groupId>
+                <artifactId>orphedomos-maven-plugin</artifactId>
+                <configuration>
+                    <imageName>pjablonski/${project.artifactId}</imageName>
```
Comment (peter-mcclonski) [inline orphedomos-test/orphedomos-test-multimodule/pom.xml]: Anonymize

File: orphedomos-test/orphedomos-test-baseline/pom.xml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+    <name>Orphedomos::Test::Baseline</name>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.technologybrewery.orphedomos</groupId>
+                <artifactId>orphedomos-maven-plugin</artifactId>
+                <configuration>
+                    <imageName>pjablonski/orphetest</imageName>
```
Comment (peter-mcclonski) [inline orphedomos-test/orphedomos-test-baseline/pom.xml]: Anonymize

**Review by peter-mcclonski — CHANGES_REQUESTED** [review]

File: orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/util/credential/CredentialUtil.java [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+        giveFullFilePermissions(newFile.getAbsolutePath());
+    }
+
+    /**
+     * Gives full read, write, and execute permissions to a file.
+     *
+     * @param filePath the path to the file
+     */
+    public static void giveFullFilePermissions(String filePath) {
+        File file = new File(filePath);
```
[the remark this answers is not in the document]
Comment (d-ryan-ashcraft) [reply orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/util/credential/CredentialUtil.java]: Fixed

File: orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/util/credential/CredentialUtil.java [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+            } else {
+                logger.info(output.toString());
+            }
+
+            int exitVal = process.waitFor();
+            if (exitVal != 0) {
+                throw new OrphedomosException("Error encountered when running bash script located at " + bashScriptPath
+                        + "\n    Can run maven build with -X to see the output of the failed script.");
+            }
+        } catch (IOException | InterruptedException e) {
```
[the remark this answers is not in the document]
Comment (d-ryan-ashcraft) [reply orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/util/credential/CredentialUtil.java]: Resolved

File: orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/util/credential/CredentialUtil.java [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+    /**
+     * Writes a given list of lines to the file located at the provided file path.
+     *
+     * @param commands the newline-delineated list of String file lines
+     * @param filePath the path to the file
+     */
+    public static void writeLinesToFile(String commands, String filePath) {
+        logger.debug("Writing lines to file located at {}.", filePath);
+
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(filePath))) {
```
[the remark this answers is not in the document]
Comment (d-ryan-ashcraft) [reply orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/util/credential/CredentialUtil.java]: Resolved

File: orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/util/credential/CredentialUtil.java [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+            if (parameters != null && parameters.length > 0) {
+                command = new String[parameters.length + 1];
+
+				System.arraycopy(parameters, 0, command, 1, parameters.length);
+            } else {
+                command = new String[1];
+            }
+            command[0] = bashScriptPath;
+
+            Process process = Runtime.getRuntime().exec(command);
```
[the remark this answers is not in the document]
Comment (d-ryan-ashcraft) [reply orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/util/credential/CredentialUtil.java]: not subject to end user invocation

File: orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/util/exec/ProcessExecutor.java [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+        if (additionalEnvironment != null) {
+            environment.putAll(additionalEnvironment);
+        }
+
+        if (platform.isWindows()) {
+            for (final Map.Entry<String, String> entry : environment.entrySet()) {
+                final String pathName = entry.getKey();
+                if (PATH_ENV_VAR.equalsIgnoreCase(pathName)) {
+                    final String pathValue = entry.getValue();
+                    environment.put(pathName, extendPathVariable(pathValue, paths));
```
[the remark this answers is not in the document]
Comment (d-ryan-ashcraft) [reply orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/util/exec/ProcessExecutor.java:129]: No delete, so not relevant here and only marginally better than how it is now

File: orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/util/credential/CredentialUtil.java [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+    /**
+     * Creates a new file at the targeted file location and gives full file
+     * permissions to the current user.
+     *
+     * @param newFile the file location
+     */
+    public static void createFileAndGivePermissions(File newFile) {
+        logger.debug("Creating new file at {}.", newFile.getAbsolutePath());
+
+        newFile = new File(newFile.getAbsolutePath());
```
[the remark this answers is not in the document]
Comment (d-ryan-ashcraft) [reply orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/util/credential/CredentialUtil.java]: Removed redundant creation of File object

File: orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/util/credential/CredentialUtil.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (d-ryan-ashcraft) [reply orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/util/credential/CredentialUtil.java]: Commands created by internal program

File: orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/util/credential/CredentialUtil.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (d-ryan-ashcraft) [reply orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/util/credential/CredentialUtil.java]: Resolved

File: orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/util/credential/CredentialUtil.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (d-ryan-ashcraft) [reply orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/util/credential/CredentialUtil.java]: Update to destroy process on completion

File: orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/util/OrphedomosException.java [hunk 10]
```diff
@@ -0,0 +1,30 @@
+package org.technologybrewery.orphedomos.util;
+public class OrphedomosException extends RuntimeException {
+    /**
+     * {@inheritDoc}
```
[the remark this answers is not in the document]
Comment (d-ryan-ashcraft) [reply orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/util/OrphedomosException.java]: Resolved - not needed unless adding more to javadoc comments

File: orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/util/credential/CredentialUtil.java [hunk 11]
```diff
@@ -0,0 +1,179 @@
...
+                command = new String[1];
+            }
+            command[0] = bashScriptPath;
+
+            Process process = Runtime.getRuntime().exec(command);
+
+            StringBuilder output = new StringBuilder();
+            String line;
+
+            BufferedReader stdInReader = new BufferedReader(new InputStreamReader(process.getInputStream()));
```
[the remark this answers is not in the document]
Comment (d-ryan-ashcraft) [reply orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/util/credential/CredentialUtil.java]: Not changing

File: orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/mojo/AbstractDockerMojo.java [hunk 12]
```diff
@@ -0,0 +1,151 @@
...
+	}
+    }
+
+    /**
+     * Encapsulates all build logic associated with this plugin goal -
+     * {@link AbstractDockerMojo#execute()} delegates to this method after
+     * performing execution handling that is common across all Mojos.
+     *
+     * @throws MojoExecutionException
+     * @throws MojoFailureException
```
[the remark this answers is not in the document]
Comment (d-ryan-ashcraft) [reply orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/mojo/AbstractDockerMojo.java]: Not real impactful, but updated to suppress implicitly moving forward

File: orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/util/exec/Platform.java [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+        String arch = System.getProperty("os.arch");
+        String version = System.getProperty("os.version");
+
+        if (arch.equals("ppc64le")) {
+            return ppc64le;
+        } else if (arch.equals("aarch64")) {
+            return arm64;
+        } else if (arch.equals("s390x")) {
+            return s390x;
+        } else if (arch.equals("arm") && version.contains("v7")) {
```
[the remark this answers is not in the document]
Comment (d-ryan-ashcraft) [reply orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/util/exec/Platform.java:26]: Default to x86 if null

File: orphedomos-test/orphedomos-test-multimodule/pom.xml (hunk 2 above)
Comment (d-ryan-ashcraft) [reply orphedomos-test/orphedomos-test-multimodule/pom.xml]: Resolved

File: orphedomos-test/orphedomos-test-baseline/pom.xml (hunk 3 above)
Comment (d-ryan-ashcraft) [reply orphedomos-test/orphedomos-test-baseline/pom.xml]: Resolved

File: orphedomos-test/orphedomos-test-no-buildkit/pom.xml (hunk 1 above)
Comment (d-ryan-ashcraft) [reply orphedomos-test/orphedomos-test-no-buildkit/pom.xml]: Resolved

File: orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/util/credential/CredentialUtil.java [hunk 14]
```diff
@@ -0,0 +1,186 @@
...
+     * Writes a given list of lines to the file located at the provided file path.
+     *
+     * @param commands the newline-delineated list of String file lines
+     * @param filePath the path to the file
+     */
+    public static void writeLinesToFile(String commands, String filePath) {
+        logger.debug("Writing lines to file located at {}.", filePath);
+
+        String noTraversalFilePath = FilenameUtils.getFullPath(filePath);
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(noTraversalFilePath))) {
```
[the remark this answers is not in the document]
Comment (d-ryan-ashcraft) [reply orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/util/credential/CredentialUtil.java]: @sonatype-lift ignore

File: orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/util/credential/CredentialUtil.java [hunk 15]
```diff
@@ -0,0 +1,186 @@
...
+
+            System.arraycopy(parameters, 0, command, 1, parameters.length);
+        } else {
+            command = new String[1];
+        }
+        command[0] = bashScriptPath;
+
+        Process process = null;
+        try {
+            process = Runtime.getRuntime().exec(command);
```
[the remark this answers is not in the document]
Comment (d-ryan-ashcraft) [reply orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/util/credential/CredentialUtil.java]: @sonatype-lift ignore

File: orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/util/credential/CredentialUtil.java (hunk 14 above)
[the remark this answers is not in the document]
Comment (d-ryan-ashcraft) [reply orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/util/credential/CredentialUtil.java]: Resolved

File: orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/util/exec/ProcessExecutor.java [hunk 16]
```diff
@@ -0,0 +1,215 @@
...
+            localEnvironment.putAll(additionalEnvironment);
+        }
+
+        if (platform.isWindows()) {
+
+            for (final Map.Entry<String, String> entry : localEnvironment.entrySet()) {
+                final String pathName = entry.getKey();
+                if (PATH_ENV_VAR.equalsIgnoreCase(pathName)) {
+                    final String pathValue = entry.getValue();
+                    localEnvironment.put(pathName, extendPathVariable(pathValue, paths));
```
[the remark this answers is not in the document]
Comment (d-ryan-ashcraft) [reply orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/util/exec/ProcessExecutor.java:130]: @sonatype-lift ignore

File: orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/util/credential/CredentialUtil.java [hunk 17]
```diff
@@ -0,0 +1,186 @@
...
+        return decryptedPassword;
+    }
+
+    /**
+     * Run the bash script found at the given location without parameters.
+     *
+     * @param bashScriptPath absolute path to the bash script
+     */
+    public static void runBashScript(String bashScriptPath) {
+        runBashScript(bashScriptPath, null, true);
```
[the remark this answers is not in the document]
Comment (d-ryan-ashcraft) [reply orphedomos-maven-plugin/src/main/java/org/technologybrewery/orphedomos/util/credential/CredentialUtil.java:75]: Resolved

**Pull request merged**